### PR TITLE
Supports ranger operators for strings in TraceQL

### LINKED ISSUE #2181: [TraceQL] Support range operators for strings
**Is your feature request related to a problem? Please describe.**
This query:
```
{span.http.status_code >= "200"}
```
will fail with this error:
```
illegal operation for the given types: span.http.status_code >= `200`
```
Support range operators (>, >= ,<=, <) for strings. Note that this should not attempt to convert strings to number values, but should instead use lexicographic ordering when comparing strings.

### PR DESCRIPTION
**Pull request by ie-pham**
**What this PR does**: Supports ranger operators for strings in TraceQL

**Which issue(s) this PR fixes**:
Fixes #2181

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by knylander-grafana** [comment]
@ie-pham Do we need to update the doc for ranger operators?

**Comment by ie-pham** [comment]
@mdisibio not sure how to do byte compare because it needs match the query type?

**Review by knylander-grafana — COMMENTED** [review]
Thank you for updating the docs! Doc changes look good.

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +39,7 @@
 * [BUGFIX] Correctly return unique spans when &&ing and ||ing spansets. [#2254](https://github.com/grafana/tempo/pull/2254) (@joe-elliott)
 * [BUGFIX] Support negative values on aggregate filters like `count() > -1`. [#2289](https://github.com/grafana/tempo/pull/2289) (@joe-elliott)
 * [BUGFIX] Support float as duration like `{duration > 1.5s}` [#2304]https://github.com/grafana/tempo/pull/2304 (@ie-pham)
+* [BUGFIX] Supports ranger operators for strings in TraceQL [#2321]https://github.com/grafana/tempo/pull/2321 (@ie-pham)
```
Comment (joe-elliott) [inline CHANGELOG.md:42]: feels like an enhancement to me

File: docs/sources/tempo/traceql/_index.md [hunk 2]
```diff
@@ -129,6 +129,12 @@ For example, to find all traces where an `http.status_code` attribute in a span
 { span.http.status_code >= 400 && span.http.status_code < 500 }
 ```
 
+This works for `http.status_code` values that are strings as well:
```
Comment (joe-elliott) [inline docs/sources/tempo/traceql/_index.md:132]: let's be clear this is lexographic ordering and not a conversion.

**Review by joe-elliott — APPROVED** [review]
nice one!

**Pull request merged**